src/btree/bt_slvg.c

```diff
@@ -194,6 +194,9 @@ __slvg_checkpoint(WT_SESSION_IMPL *session, WT_REF *root)
      * We may not have found any pages during salvage and there's no tree to flush.
      */
     if (root->page != NULL) {
+        /* Clear the saved checkpoint information. */
+        __wt_meta_saved_ckptlist_free(session);
+
```

Comment:
I'm confused by this call, what's the path where it's necessary? I would have expected the close to clear this, and it not to be needed here.

Reply:
It is my ignorance of the salvage code and my attempt at being extra cautious. I will replace this `__wt_meta_saved_ckptlist_free` call with assert confirming that `btree->ckpt` is `NULL`

Reply:
Replaced with assert, marking the conversation resolved.